src/main/java/dev/morling/onebrc/CalculateAverage_royvanrijn.java

```diff
@@ -250,72 +267,92 @@ private static long compilePattern(final byte value) {
-     *
-     * So I've written an extremely simple linear probing hashmap that should work well enough.
+     * Extremely simple linear probing hashmap that should work well enough.
      */
     class MeasurementRepository {
-        private int tableSize = 1 << 20; // large enough for the contest.
-        private int tableMask = (tableSize - 1);
-
-        private MeasurementRepository.Entry[] table = new MeasurementRepository.Entry[tableSize];
+        private final int tableSize = 1 << 18; // large enough for the contest.
+        private final int tableMask = (tableSize - 1);
+
+        private final Entry[] table = new Entry[tableSize];
+        /**
+         * Separated hashtable, keeps memory local, idea of from franz1981.
+         */
+        private final int[] hashTable = new int[tableSize];
+
+        static final class Entry {
+            private final long[] data;
+            private final String city;
+            private int min, max, count;
+            private long sum;
+
+            Entry(long[] data, String city) {
+                this.data = data;
+                this.city = city;
+                this.min = 1000;
+                this.max = -1000;
+            }
 
-        record Entry(long address, long[] data, int length, int hash, String city, Measurement measurement) {
+            public void updateWith(int measurement) {
+                min = min(min, measurement);
+                max = max(max, measurement);
+                sum += measurement;
+                count++;
+            }
 
-            @Override
-            public String toString() {
-                return city + "=" + measurement;
+            public Measurement measurement() {
+                return new Measurement(min, max, count, sum);
             }
         }
 
-        public void update(long address, long[] data, int length, int hash, int temperature) {
+        public void update(final long address, final long[] data, final int length, final int hash, final int temperature) {
+
+            final int dataLength = (length >> 3) + 1;
 
-            int dataLength = length >> 3;
             int index = hash & tableMask;
-            MeasurementRepository.Entry tableEntry;
-            while ((tableEntry = table[index]) != null
-                    && (tableEntry.hash != hash || tableEntry.length != length || !arrayEquals(tableEntry.data, data, dataLength))) { // search for the right spot
+            Entry tableEntry = null;
+
+            // Find the entry:
+            while (true) {
+                int hashTableEntry;
+                if ((hashTableEntry = hashTable[index]) == 0) {
+                    // Slot is empty
+                    break;
+                }
+                else if (hashTableEntry == hash) {
+                    tableEntry = table[index];
+                    // Match the entire long[] with all name-bytes:
+                    if (Arrays.mismatch(tableEntry.data, 0, dataLength, data, 0, dataLength) < 0) {
```

Comment:
note:

graal JIT autovectorization does great here, including inlining the equals - mismatch is it not inlined, but it's a bit more compact than what is produced by Graal JIT for the long by long original method

Reply:
> maybe some tuning of the inlining heuristics could help

I don't know if `mismatch` could be inlined, really; it create a stub to be called, separated. At least is what I can get by reading the assembly, probably only @thomaswue know (or other Graal committers)?